armory/scenarios/poisoning_witches_brew.py

```diff
@@ -340,12 +342,51 @@ def load_dataset(self, eval_split_default="test"):
         self.i = -1
 
     def load_metrics(self):
-        self.non_trigger_accuracy_metric = metrics.MetricList("categorical_accuracy")
-        self.trigger_accuracy_metric = metrics.MetricList("categorical_accuracy")
+        self.hub.connect_meter(
+            Meter(
+                "mean_accuracy_non_trigger_images",
+                metrics.get_supported_metric("categorical_accuracy"),
+                "scenario.y[non-trigger]",
+                "scenario.y_pred[non-trigger]",
+                final=np.mean,
+                final_name="mean_accuracy_non_trigger_images",
+                record_final_only=False,
+            )
+        )
+        self.hub.connect_meter(
+            Meter(
+                "accuracy_trigger_images",
+                metrics.get_supported_metric("categorical_accuracy"),
+                "scenario.y[trigger]",
+                "scenario.y_pred[trigger]",
+                final=np.mean,
+                final_name="mean_accuracy_trigger_images",
+                record_final_only=False,
+            )
+        )
+
+        per_class_accuracy = metrics.get_supported_metric("per_class_accuracy")
+        self.hub.connect_meter(
+            Meter(
+                "sample_non_trigger_test_accuracy_per_class",
+                metrics.get_supported_metric("identity_unzip"),
+                "scenario.y[non-trigger]",
+                "scenario.y_pred[non-trigger]",
+                final=lambda x: per_class_accuracy(*metrics.identity_zip(x)),
+                final_name="mean_accuracy_per_class_non_trigger_images",
+                record_final_only=True,
+            )
+        )
+
+        # TODO:
+        # log.info(
+        #     f"Accuracy on non-trigger images: {self.non_trigger_accuracy_metric.mean():.2%}"
+        # )
+
+        # log.info(
+        #     f"Accuracy on trigger images: {self.trigger_accuracy_metric.mean():.2%}"
+        # )
 
-        self.benign_test_accuracy_per_class = (
-            {}
-        )  # store accuracy results for each class
         if self.config["adhoc"].get("compute_fairness_metrics", False):
             self.fairness_metrics = FairnessMetrics(
                 self.config["adhoc"], self.use_filtering_defense, self
```

Comment:
This might be missing a couple things, just comparing to load_metrics() in poison.py:
- add the filter perplexity metric (if self.use_filtering_defense): `self.fairness_metrics.add_filter_perplexity()`
- set up the results writer: 
```
self.results_writer = ResultsWriter(sink=None)
self.hub.connect_writer(self.results_writer, default=True)
```

Reply:
Good catch!

Reply:
Load_metrics is also missing the tpr/fpr metric:
```
        if self.use_filtering_defense:
            # Filtering metrics
            self.hub.connect_meter(
                Meter(
                    "filter",
                    metrics.get_supported_metric("tpr_fpr"),
                    "scenario.poisoned",
                    "scenario.removed",
                )
            )
```